airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/write/WriteOperation.kt

```diff
@@ -74,9 +77,21 @@ class WriteOperation(
 @Factory
 class InputStreamProvider {
     @Singleton
-    @Secondary
-    @Requires(property = Operation.PROPERTY, value = "write")
-    fun make(): InputStream {
-        return System.`in`
+    @Named("inputStream")
+    fun makeWrite(
+        @Value("\${${Operation.PROPERTY}}") operation: String,
+        @Named("checkInputStreamPipe") pipe: PipedOutputStream?,
+    ): InputStream =
+        if (operation == "check") {
```

Comment:
Nit: We'll probably want a set of constants for the operation names

Reply:
yeahhhh. I'll either do that in a separate pr, or write a quick ticket to do it - didn't want to pollute this PR with a big find+replace